sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1335,4 +1361,42 @@ export class BlobClient extends StorageClient {
     (response as any).blobDownloadStream = undefined;
     return response;
   }
+
+  private getBlobAndContainerNamesFromUrl(
+    url: string
+  ): { blobName: string; containerName: string } {
+    url = url.split("?")[0]; // removing the sas part of url if present
```

Comment:
[@jeremymeng's comment from HarshaNalluru/pull/21 ]
There are some usage of URLBuilder parsing urls. Would it help here?

Reply:
There is a method to append but didn't find any to strip off.

Though I see some functions that are not exposed but used internally in core-http library that might be helpful in this particular case.

https://github.com/Azure/azure-sdk-for-js/blob/4c4a859da1135f3a27e476c8d5534a74487e4970/sdk/core/core-http/lib/url.ts#L568-L574